docs/DeveloperGuide.adoc

```diff
@@ -154,89 +145,86 @@ Classes used by multiple components are in the `seedu.addressbook.commons` packa
 This section describes some noteworthy details on how certain features are implemented.
 
 // tag::undoredo[]
-=== [Proposed] Undo/Redo feature
-==== Proposed Implementation
+=== Analyse Password
+The following class diagram is the current structure of the `Analyser` component, using `DictionaryAnalyser` as an example:
 
-The undo/redo mechanism is facilitated by `VersionedAddressBook`.
-It extends `AddressBook` with an undo/redo history, stored internally as an `addressBookStateList` and `currentStatePointer`.
-Additionally, it implements the following operations:
+image::AnalyserClassDiagram.png[]
```

Comment:
The diagram can be resized.
It appears too verbose.
Each component could be described individually.
Missing figure caption.